### PR TITLE
Generate relase notes

### LINKED ISSUE #5: Release process for publishing to JetBrains plugin marketplace
should include credentials management through github secrets

### PR DESCRIPTION
**Pull request by vgramer**
# Description
This PR automates the generation of the release notes and the publish of the GitHub release and plugin on Jetbrains marketplace. 

The process is:
1. contributors make PR as usual and add special labels `kind/<the appropriate kind>`, `area/<the appropriate area>` and write note inside block `release-notes` in PR description
1.every time a PR is merged on master, the action `upset_release_notes_pr_on_push_on_master.yml` is triggered  and create or update the PR to update the changelog with the new release-notes
1. ensure release-notes PR looks good. Edit the corresponding PR(s) if needed.
1. edit the version in release-notes PR and merge it:
   1. edit the title of last release notes to change `NEXT_VERSION` by the actual version (must matches `v[0-9]+.[0-9]+.[0-9]+)`), you can do it in GitHub interface.
   1. merge pr using **squashing** strategy
1. tag the `master` branch with the actual version (this will create a Github release and publish the plugin on Jetbrains store)

To know more about are release-notes are gathered please read  [docs/devel/releasing/release-notes-generation.md](https://github.com/vgramer/opa-idea-plugin/blob/generate-relase-notes/docs/devel/releasing/release-notes-generation.md)

Fixes #5

# Release notes 
```release-notes
automate release process, create CHANGLOG.md with update the previous releases notes
```

# Special notes for your reviewer

the way to generate the plugin version number has changed. 
Now it's: `${pluginversion}-${platformVersion}${chanel}` where:
      * `pluginversion` comes from gradle property. (this prop will be overwritten by github action with the tag of the release)
      * `platformVersion`= 203 or 211
      * `chanel `=  ``  (ie blank )for release,  `-dev` for testing

### CODE REVIEW

File: .github/workflows/release_and_pulbish_plugin_on_tag.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+name: publish plugin on JB and create release on GH
```
Comment (asadali) [inline .github/workflows/release_and_pulbish_plugin_on_tag.yml:1]: typo in file name. `s/pulbish/publish`

File: .github/workflows/release_and_pulbish_plugin_on_tag.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        uses: actions/download-artifact@v2
+        id: download
+        with:
+          name: plugin-artifact
+          path: ./artifacts/
+
+      - name: create or update release note PR
+        env:
+          # Required for the `hub` CLI
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (asadali) [inline .github/workflows/release_and_pulbish_plugin_on_tag.yml:66]: is this secret present by default? (i don't see it in the settings/secrets section)

File: docs/devel/releasing/releasing_plugin.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Table of Content
+<!-- toc -->
+- [How to release the plugin](#how-to-release-the-plugin)
```
Comment (asadali) [inline docs/devel/releasing/releasing_plugin.md:3]: thanks for writing this up!

**Review by asadali — APPROVED** [review]
LGTM! thanks for the detailed work on improving the process.

File: .github/workflows/release_and_pulbish_plugin_on_tag.yml (hunk 2 above)
Comment (vgramer) [reply .github/workflows/release_and_pulbish_plugin_on_tag.yml:66]: yes it provided by github by default

File: .github/workflows/release_and_pulbish_plugin_on_tag.yml (hunk 1 above)
Comment (vgramer) [reply .github/workflows/release_and_pulbish_plugin_on_tag.yml:1]: done

**Pull request merged**